hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/resolver/MembershipNamenodeResolver.java

```diff
@@ -488,7 +488,7 @@ public void setRouterId(String router) {
    * @param listObserversFirst Observer read case, observer NN will be ranked first
    */
   @Override
-  public synchronized void rotateCache(
+  public void rotateCache(
       String nsId, FederationNamenodeContext namenode, boolean listObserversFirst) {
```

Comment:
cacheNS instance of ConcurrentHashMap, so the compute method is atomic, we can delete synchronized of the method rotateCache